### PR TITLE
Fix #1793 and test creating widgets for model fields

### LINKED ISSUE #1793: Add support for PositiveBigIntegerField
**Describe the bug**

If model has PositiveBigIntegerField(null=True) resource has autogenerated field with default widget and imported dataset contains empty cells - there is import error in saving imported instance. Probably it happens because `ModelResource.WIDGETS_MAP` does not contains widget class for PositiveBigIntegerField.

```
Field 'field_example' expected a number but got ''. OrderedDict({'id': 1, 'field_example': ''})
```

**To Reproduce**

1. Create model with PositiveBigIntegerField(null=True)  and Resource for this
```python
from django.db import models
from import_export import resources


class ExampleModel(models.Model):
    field_example = models.PositiveBigIntegerField(null=True)

class ExampleResource(resources.ModelResource):
    class Meta:
        model = ExampleModel

```
2. Create some instances
```python
ExampleModel.objects.create()
ExampleModel.objects.create(field_example=123)
```
3. Export data to dataset
```python
resource = ExampleResource()
dataset = resource.export()
```

4. Clear db
```python
ExampleModel.objects.all().delete()
```
5. Import this dataset and check result
```python
result = resource.import_data(dataset)
self.assertFalse(result.has_errors())
```

**Versions (please complete the following information):**
 - Django Import Export: 3.3.8
 - Python 3.12
 - Django 4.2.11

**Expected behavior**
Dataset imported correctly

**Additional context**

Maybe create test for check that all django model fields are supported(maybe except FileField/ImageField)? It looks like `PositiveSmallIntegerField` is missing to.

** Full test to reproduce **

```python
# models.py
from django.db import models

class ExampleModel(models.Model):
    field_example = models.PositiveBigIntegerField(null=True)

```

```python
# tests.py

from django.test import TestCase
from import_export import resources
from import_export.results import Error

from example_app.models import ExampleModel

class ExampleResource(resources.ModelResource):
    class Meta:
        model = ExampleModel


class TestImportExportBug(TestCase):
    def setUp(self) -> None:
        ExampleModel.objects.create()
        ExampleModel.objects.create(field_example=123)

    def test_bug(self):
        resource = ExampleResource()
        dataset = resource.export()

        ExampleModel.objects.all().delete()

        result = resource.import_data(dataset)

        self.assertFalse(
            result.has_errors(),
            msg=(
                    "\n".join(self._format_error(error) for error in result.base_errors)
                    + "\n\n"
                    + "\n".join(
                "\n".join(self._format_error(error) for error in errors) for _, errors in result.row_errors()
            )
            )
        )

    def _format_error(self, error: Error) -> str:
        return f"{error.error} {error.row}\n{error.traceback}"
```

### PR DESCRIPTION
**Pull request by mishka251**
**Problem**

Issue #1793. Model resource does not creating widgets for some model field classes.

**Solution**

- [x] create test
- [x] create solution
- [ ] write docs. Not needed

Add a test that checks whether a special widget is being created for each of the types of model fields. I hope it can help to find similar problems with another fields.

**Acceptance Criteria**

Have you written tests? Yes

Have you included screenshots of your changes if applicable? Not needed, no UI changes

Did you document your changes? Test has comments and doc strings. There are no significant changes to the public API.

### CODE REVIEW

**Comment by matthewhegarty** [comment]
Thanks for this, and the comprehensive tests.

I would like this to go into the forthcoming v4 release, so is it possible you could rebase onto `release-4`?

It would ideal if you could update `CHANGELOG.rst` and feel free to add your name to `AUTHORS`.

**Comment by mishka251** [comment]
Rebased and change target to `release-4`.

Changelog and authors updated.

File: tests/core/tests/test_model_resource_fields_generate_widgets.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+from ..models import WithPositiveIntegerFields
+
+
+class ExampleResource(ModelResource):
+    class Meta:
+        model = WithPositiveIntegerFields
+
+
+class TestImportExportBug(TestCase):
```
Comment (matthewhegarty) [inline tests/core/tests/test_model_resource_fields_generate_widgets.py:22]: Please could you rename this to a more descriptive term?  Perhaps `TestFieldWidgetMapping`.

File: tests/core/tests/test_model_resource_fields_generate_widgets.py [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+         _get_django_fields_for_check_widget
+        """
+        expected_not_presented_fields = {
+            contenttype_fields.GenericRelation,
+            models.ForeignObject,
+            postgres_search.SearchQueryField,
+            postgres_search.SearchVectorField,
+            RelatedField,
+        }
+        if django.VERSION >= (4, 1):
```
Comment (matthewhegarty) [inline tests/core/tests/test_model_resource_fields_generate_widgets.py:123]: Our minimum supported version is 4.2 so perhaps this check isn't necessary?

File: tests/core/tests/test_model_resource_fields_generate_widgets.py [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+            expected_not_presented_fields |= {contenttype_fields.GenericForeignKey}
+        return expected_not_presented_fields
+
+    def _get_all_django_model_field_subclasses(self):
+        """
+        returns list of classes - all subclasses for django.db.models.field.Field
+        """
+        return self._collect_all_clas_children(models.Field)
+
+    def _collect_all_clas_children(self, cls):
```
Comment (matthewhegarty) [inline tests/core/tests/test_model_resource_fields_generate_widgets.py:139]: should this be `cls` not `clas`?

**Review by matthewhegarty — CHANGES_REQUESTED** [review]
Thanks so much for submitting this PR.  I've reviewed and have some minor comments.

**Comment by mishka251** [comment]
When I was solving conflicts with `release-4`, I wrote my changes in `4.0.0` is this correct?

**Comment by matthewhegarty** [comment]
Yes, all good, thanks

**Review by matthewhegarty — APPROVED** [review]

**Pull request merged**